Sources/MacroToolkit/Modifiers/Modifier.swift

```diff
@@ -0,0 +1,23 @@
+import SwiftSyntax
+
+/// A protocol for modifiers in Swift that are represented by `TokenKind`.
+public protocol Modifier: RawRepresentable where RawValue == TokenKind {
```

Comment:
I can see a concrete `Modifier` enum being desirable in the future (just like the other protocols have corresponding concrete enums), so it might be best to rename this to `ModifierProtocol` to avoid breaking changes when the time comes.